### PR TITLE
[Refactor] Unify all authentication interfaces to PrivilegeChecker

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 205 / 269 (76.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/AccessControl.java|0|29|00.00%|[26, 31, 35, 40, 44, 48, 52, 56, 57, 60, 64, 65, 68, 73, 77, 78, 81, 86, 87, 90, 94, 99, 103, 110, 114, 118, 122, 127, 131]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|3|66.67%|[770]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeActions.java|14|20|70.00%|[145, 146, 159, 166, 184, 353]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|61|76|80.26%|[63, 64, 65, 150, 151, 176, 177, 208, 227, 235, 236, 239, 240, 242, 243]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerVisitor.java|122|135|90.37%|[350, 559, 571, 583, 595, 757, 879, 966, 1058, 1091, 1387, 1417, 1566]|
|:large_blue_circle:|com/starrocks/privilege/Authorizer.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeActions.java [hunk 1]
```diff
@@ -33,7 +34,7 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public class PrivilegeActions implements SystemAccessControl {
+public class PrivilegeActions implements AccessControl {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeActions.java:37]: Shall we unify the naming, `AccessControl` and `AstVisitor ` are the base classes, so we will have `NativeAccessControl`(to replace `PrivilegeActions`) and `NativePrivilegeChecker`, `NativePrivilegeChekcerVisitor`. Later we can have `RangerAccessControl`, `RangerPrivilegeChecker`...

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeActions.java:37]: Currently, `PrivilegeXXX` without prefix is too general and may confuse the developers.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 50 / 92 (54.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/GetLoadInfoAction.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/privilege/AccessControl.java|0|29|00.00%|[26, 31, 35, 40, 44, 48, 52, 56, 57, 60, 64, 65, 68, 73, 77, 78, 81, 86, 87, 90, 94, 99, 103, 110, 114, 118, 122, 127, 131]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadFunctionalExprProvider.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|1|2|50.00%|[228]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|7|12|58.33%|[364, 885, 906, 922, 1655]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|20|75.00%|[769, 770, 1468, 1682, 1737]|
|:large_blue_circle:|com/starrocks/http/BaseAction.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectScheduler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadFunctionalExprProvider.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 50 / 92 (54.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/GetLoadInfoAction.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/privilege/AccessControl.java|0|29|00.00%|[26, 31, 35, 40, 44, 48, 52, 56, 57, 60, 64, 65, 68, 73, 77, 78, 81, 86, 87, 90, 94, 99, 103, 110, 114, 118, 122, 127, 131]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadFunctionalExprProvider.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|1|2|50.00%|[228]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|7|12|58.33%|[364, 885, 906, 922, 1655]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|20|75.00%|[769, 770, 1468, 1682, 1737]|
|:large_blue_circle:|com/starrocks/http/BaseAction.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectScheduler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadFunctionalExprProvider.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 305 / 389 (78.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/GetLoadInfoAction.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/privilege/AccessControl.java|0|29|00.00%|[26, 31, 35, 40, 44, 48, 52, 56, 57, 60, 64, 65, 68, 73, 77, 78, 81, 86, 87, 90, 94, 99, 103, 110, 114, 118, 122, 127, 131]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadFunctionalExprProvider.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|1|2|50.00%|[228]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|7|12|58.33%|[364, 885, 906, 922, 1655]|
|:large_blue_circle:|com/starrocks/privilege/NativeAccessControl.java|15|21|71.43%|[145, 146, 159, 166, 184, 353]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|20|75.00%|[769, 770, 1468, 1682, 1737]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Authorizer.java|104|127|81.89%|[36, 63, 64, 65, 150, 151, 176, 177, 178, 208, 212, 227, 235, 236, 237, 239, 240, 241, 242, 243, 244, 245, 251]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AuthorizerStmtVisitor.java|130|143|90.91%|[350, 559, 571, 583, 595, 757, 879, 966, 1058, 1091, 1387, 1417, 1566]|
|:large_blue_circle:|com/starrocks/http/BaseAction.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectScheduler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/AccessControlProvider.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadFunctionalExprProvider.java|1|1|100.00%|[]|

**Pull request merged**